bouncer/tests/signed_runtime_call.ts

```diff
@@ -341,11 +367,141 @@ async function testEvmEip712Encoding(logger: Logger) {
     .send();
 }
 
+async function testSpecialLpDeposit(logger: Logger, asset: Asset) {
+  await using chainflip = await getChainflipApi();
+
+  const initialFlipToBeSentToGateway = (
+    await chainflip.query.swapping.flipToBeSentToGateway()
+  ).toJSON() as number;
+
+  logger.info('Setting up a broker account');
+  const brokerUri = `//BROKER_SPECIAL_DEPOSIT_CHANNEL`;
+  const broker = createStateChainKeypair(brokerUri);
+  await setupBrokerAccount(logger, brokerUri);
+
+  const evmWallet = await createEvmWallet();
+  const evmScAccount = externalChainToScAccount(evmWallet.address);
+  logger.info('evmScAccount for special LP deposit channel:', evmScAccount);
+  const evmNonce = (await chainflip.rpc.system.accountNextIndex(evmScAccount)).toNumber();
+  const refundAddress = await newAssetAddress(asset, brokerUri + Math.random() * 100);
+
+  let addressBytes;
+
+  if (asset === 'Btc') {
+    // In prod we should encode the BTC with the adequate encoding, this is to keep it simple
+    addressBytes = new Uint8Array(Buffer.from(refundAddress, 'utf-8'));
+  } else {
+    addressBytes = new Uint8Array(Buffer.from(refundAddress.slice(2), 'hex'));
+  }
+
+  const remarkData = remarkDataCodec.enc({ tag: shortChainFromAsset(asset), value: addressBytes });
+
+  const call = chainflip.tx.system.remark(Array.from(remarkData));
+  const hexRuntimeCall = u8aToHex(chainflip.createType('Call', call.method).toU8a());
+
+  const response = await chainflip.rpc(
+    'cf_encode_non_native_call',
+    hexRuntimeCall,
+    blocksToExpiry,
+    evmNonce,
+    { Eth: 'Eip712' },
+  );
+
+  const [eipPayload, transactionMetadata] = encodeNonNativeCallResponseSchema.parse(response);
+  const parsedPayload = encodedNonNativeCallSchema.parse(eipPayload);
+  const { domain, types, message } = parsedPayload.Eip712;
+  delete types.EIP712Domain;
+
+  const evmSignatureEip712 = await evmWallet.signTypedData(domain, types, message);
+
+  const nonce = await chainflip.rpc.system.accountNextIndex(broker.address);
+  await chainflip.tx.swapping
+    .requestAccountCreationDepositAddress(
+      {
+        Ethereum: {
+          signature: evmSignatureEip712,
+          signer: evmWallet.address,
+          sigType: 'Eip712',
+        },
+      },
+      {
+        nonce: transactionMetadata.nonce,
+        expiryBlock: transactionMetadata.expiry_block,
+      },
+      asset,
+      0,
+      { [shortChainFromAsset(asset).toLowerCase()]: refundAddress },
+    )
+    .signAndSend(broker, { nonce }, handleSubstrateError(chainflip));
+
+  logger.info('Opening special deposit channel and depositing..');
+
+  const eventResult = await observeEvent(logger, 'swapping:AccountCreationDepositAddressReady', {
+    test: (event) =>
+      event.data.requestedBy === broker.address && event.data.requestedFor === evmScAccount,
+  }).event;
```

Comment:
I think we should add `historicalCheckBlocks: 10` here, otherwise we have a race condition between submitting the extrinsic and observing the event, AFAICT.